lib/Message.cc

```diff
@@ -136,6 +140,15 @@ void Message::setMessageId(const MessageId& messageID) const {
     return;
 }
 
+const int64_t Message::getIndex() const {
+    if (!impl_) {
+        return -1;
+    } else {
+        // casting uint64_t to int64_t, server definition ensures that's safe
+        return static_cast<int64_t>(impl_->brokerEntryMetadata.index());
```

Comment:
I think we should check `has_index()` as well because `index` is an optional field.

```c++
if (!impl_ || !impl_->has_index()) {
    return -1;
```

Reply:
@BewareMyPower Fixed